chain/network/src/routing.rs

```diff
@@ -305,6 +315,10 @@ impl RoutingTable {
         &self.raw_graph.source
     }
 
+    pub fn reachable_peers(&self) -> Vec<PeerId> {
```

Comment:
`-> impl Iterator<Item = PeerId>` should save us cloning and a vector allocation on every ping.

Reply:
Solving the problem of cloning + vector allocation is no possible. We can't have reference to Routing Table since we are requiring mutable access later using peers retrieved from here. What can be done is moving all the logic entirely inside RoutingTable but I think it is unnecessary, since pings will be done 1 every 20 seconds right now, and can be increased on demand.

Also, added that this routine is only called if compiled with `metric_recorder` feature enabled.

Reply:
> Also, added that this routine is only called if compiled with `metric_recorder` feature enabled.

Can we make it explicit with `#[cfg(feature = "metric_recorder")]`?

Reply:
Added your suggestion to `reachable_peers` and moved clone+vector to `ping_all_peers` which is already behind the feature flag.